### PR TITLE
Return a Reader from the Exec function

### PR DESCRIPTION
**Pull request by johanoskarsson**
This should allow the user to read the logs of the command execution. I've tested this quickly locally and it seems to work as expected.
Pushing this so I can get some CI feedback.
I believe this is similar to what was asked for in #126

### CODE REVIEW

File: logconsumer_test.go
Comment (mdelapenya) [inline logconsumer_test.go:249]: Is this correct?
```suggestion
		i, _, err := dind.Exec(ctx, []string{"wget", "--spider", "localhost:" + port.Port()})
```

**Comment by johanoskarsson** [comment]
Thanks I had them the wrong way around first!

**Comment by mdelapenya** [comment]
Hey @johanoskarsson, I took a look at your PR and it seems it’s outdated (created before the commit that fixed the netlify deployment in main):

> This branch is 2 commits ahead, 13 commits behind testcontainers/testcontainers-go:main

Could you rebase and push? Thank you in advance for your work here

**Review by mdelapenya — APPROVED** [review]
LGTM! Let's wait for the CI

Thanks

**Comment by mdelapenya** [comment]
NIT: before merging this, could you add a test (or an assertion in the existing tests) where we are consuming the Reader? I'm checking that we are always omitting it in the current tests, and would like to see it covered.

**Comment by mdelapenya** [comment]
BTW, I'm tagging this PR as breaking-change because we are changing the signature of an exposed method. Because we are not in 1.0.0, it's fine to break it :)

**Review by gianarb — APPROVED** [review]

**Pull request merged**
